src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java

```diff
@@ -29,24 +29,44 @@ public DeleteAction(String... paths) {
     @Override
     public long invoke(File pipelineDir, String version, boolean dryRun) {
         File versionDir = new File(pipelineDir.getAbsolutePath() + "/" + version);
-        long size = FileUtils.sizeOfDirectory(versionDir);
-
-        if (dryRun) {
-            LOG.info("[DRY RUN] Will remove " + versionDir.getAbsolutePath() + ", size = " + FileUtils.byteCountToDisplaySize(size));
-        } else {
-            LOG.info("Deleting " + versionDir.getAbsolutePath() + ", size = " + FileUtils.byteCountToDisplaySize(size));
-            try {
-                deleteDirectory(versionDir);
-            } catch (IOException e) {
-                LOG.error(e.getMessage(), e);
-                throw new RuntimeException(e);
+        long size = sizeOfDirectory(versionDir);
+
+        if (size > 0) {
```

Comment:
A failing test reveals that this check leaves 0 size files in the folders (not a big issue, but messy).
Will fix.